bisection.go

```diff
@@ -4,15 +4,30 @@
 
 package optimize
 
-import "math"
+import (
+	"math"
+	"os"
+
+	"github.com/gonum/floats"
+)
 
 // Bisection is a Linesearcher that uses a bisection to find a point that
 // satisfies the strong Wolfe conditions with the given gradient constant and
 // function constant of zero. If GradConst is zero, it will be set to a reasonable
 // value. Bisection will panic if GradConst is not between zero and one.
+// EqTol is used to smooth out floating point noise when testing the Wolfe
+// conditions. The Wolfe condition will be considered met if the function value
+// is less than or within a tolerance of the starting value of the line
```

Comment:
This is not true, we compare to the value on the left side of the interval (Is it even correct? The original code compared to the lowest value found).

There is not a suitable line to put the comment on, but `b.initF` field and `minF` variable are not used in this version, is it on purpose?